### PR TITLE
programmer-calculator: init at 2.1

### PR DESCRIPTION
**Pull request by cjab**
###### Motivation for this change

programmer-calculator is a terminal calculator for programmers.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Comments

One concern I have is that this package contains a binary named `pcalc`. A _different_ package named `pcalc` already exists in nixpkgs that also contains a binary with this name. I noticed `priority` in metadata which I think would prioritize one of these over the other? But it wasn't clear to me which is higher priority. Another option is to just rename this binary from `pcalc` -> `programmer-calculator` or something similar. I'm not sure what the usual approach to these sorts of problems is. Any guidance is welcome! Thank you! 😄

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 127355` at 5151f2cc run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>programmer-calculator</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/science/math/programmer-calculator/default.nix:16:3:
```
   |
16 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/applications/science/math/programmer-calculator/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    install -Dm 555 pcalc -t "$out/bin"
+  '';
+
+  meta = with lib; {
+    description = "A terminal calculator for programmers";
+    longDescription = ''
+      Terminal calculator made for programmers working with multiple number
+      representations, sizes, and overall close to the bits
+    '';
+    homepage = "http://alt-romes.github.io/programmer-calculator";
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/programmer-calculator/default.nix:26]:
```suggestion
    homepage = "https://alt-romes.github.io/programmer-calculator";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/science/math/programmer-calculator/default.nix (hunk 1 above)
Comment (cjab) [reply pkgs/applications/science/math/programmer-calculator/default.nix]: Good catch, fixed. Thank you!

**Comment by lrworth** [comment]
Build fails on macOS:

```
building '/nix/store/4ysqcig9hv69z232hslhw0cxjhsk4ljg-programmer-calculator-2.1.drv'...
unpacking sources
unpacking source archive /nix/store/wlwlah3xxfhwbm76sjzdm0r5hf29rk53-source
source root is source
patching sources
configuring
no configure script, doing nothing
building
build flags: SHELL=/nix/store/xxfxh4npdlp0w5winp2f17ji1yxdivn3-bash-4.4-p23/bin/bash
gcc -Wall -Wextra -g -Werror=missing-declarations -Werror=redundant-decls -Iinclude -c src/draw.c -o src/draw.o
/nix/store/xxfxh4npdlp0w5winp2f17ji1yxdivn3-bash-4.4-p23/bin/bash: gcc: command not found
make: *** [Makefile:41: src/draw.o] Error 127
builder for '/nix/store/4ysqcig9hv69z232hslhw0cxjhsk4ljg-programmer-calculator-2.1.drv' failed with exit code 2
error: build of '/nix/store/4ysqcig9hv69z232hslhw0cxjhsk4ljg-programmer-calculator-2.1.drv' failed
```

Adding `ghc` to the `buildInputs` fixes it for me.

**Comment by cjab** [comment]
> Build fails on macOS:
> 
> ```
> building '/nix/store/4ysqcig9hv69z232hslhw0cxjhsk4ljg-programmer-calculator-2.1.drv'...
> unpacking sources
> unpacking source archive /nix/store/wlwlah3xxfhwbm76sjzdm0r5hf29rk53-source
> source root is source
> patching sources
> configuring
> no configure script, doing nothing
> building
> build flags: SHELL=/nix/store/xxfxh4npdlp0w5winp2f17ji1yxdivn3-bash-4.4-p23/bin/bash
> gcc -Wall -Wextra -g -Werror=missing-declarations -Werror=redundant-decls -Iinclude -c src/draw.c -o src/draw.o
> /nix/store/xxfxh4npdlp0w5winp2f17ji1yxdivn3-bash-4.4-p23/bin/bash: gcc: command not found
> make: *** [Makefile:41: src/draw.o] Error 127
> builder for '/nix/store/4ysqcig9hv69z232hslhw0cxjhsk4ljg-programmer-calculator-2.1.drv' failed with exit code 2
> error: build of '/nix/store/4ysqcig9hv69z232hslhw0cxjhsk4ljg-programmer-calculator-2.1.drv' failed
> ```
> 
> Adding `ghc` to the `buildInputs` fixes it for me.

Thank you, fixed!

File: pkgs/applications/science/math/programmer-calculator/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  version = "2.1";
+
+  src = fetchFromGitHub {
+    owner = "alt-romes";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1vvpbj24ijl9ma0h669n9x0z1im3vqdf8zf2li0xf5h97b14gmv0";
+  };
+
+  buildInputs = [ gcc ncurses ];
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/programmer-calculator/default.nix:14]:
```suggestion
  buildInputs = [ ncurses ];
```
part of the stdenv.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/science/math/programmer-calculator/default.nix (hunk 2 above)
Comment (lrworth) [reply pkgs/applications/science/math/programmer-calculator/default.nix:14]: This was added to make it build on macOS as per my comment above. What is the correct way to fix this problem?

Comment (SuperSandro2000) [reply pkgs/applications/science/math/programmer-calculator/default.nix:14]: darwin by default uses clang. To force gcc please use gccStdenv.

Comment (cjab) [reply pkgs/applications/science/math/programmer-calculator/default.nix]: Thanks! Switched to `gccStdenv`.

**Pull request merged**